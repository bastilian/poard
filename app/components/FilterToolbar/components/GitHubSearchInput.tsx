import { TextInput } from '@patternfly/react-core/dist/umd/react-core';

interface GitHubSearchInputProps {
  onChange?: () => void
}

const GitHubSearchInput: React.FC<GitHubSearchInputProps> = ({ onChange }) => {

  const searchOnChange = () => {
    onChange?.();
  };

  return (
    <TextInput
      iconVariant="search"
      type="search"
      onChange={searchOnChange}
    />
  );
};

export default GitHubSearchInput;
