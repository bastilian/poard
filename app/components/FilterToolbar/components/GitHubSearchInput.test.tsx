import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import GitHubSearchInput from './GitHubSearchInput';

describe('GitHubSearchInput', () => {
  it('should render', () => {
    const { asFragment } = render(<GitHubSearchInput />);
    expect(asFragment()).toMatchSnapshot();
  });
});
