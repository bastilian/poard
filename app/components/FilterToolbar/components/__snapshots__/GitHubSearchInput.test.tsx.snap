// Vitest Snapshot v1

exports[`GitHubSearchInput > should render 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    class="pf-c-form-control pf-m-search"
    data-ouia-component-id="OUIA-Generated-TextInputBase-1"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    type="search"
    value=""
  />
</DocumentFragment>
`;
